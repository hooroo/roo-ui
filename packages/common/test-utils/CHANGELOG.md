# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
