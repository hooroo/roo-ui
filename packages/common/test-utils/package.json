{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
