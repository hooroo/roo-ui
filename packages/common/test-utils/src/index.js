export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
