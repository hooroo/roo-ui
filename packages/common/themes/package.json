{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
