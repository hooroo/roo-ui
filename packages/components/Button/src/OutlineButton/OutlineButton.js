import { themeGet, buttonStyle } from 'styled-system';
import Button from '..';

const getBackground = props => (props.bg
  ? themeGet(`colors.${props.bg}`, props.bg)(props)
  : buttonStyle(props).backgroundColor);

export default Button.extend`
  background-color: transparent;
  border-color: ${getBackground};
  transition: none;

  &:not(:hover), &:disabled {
    color: ${getBackground};
  }

  &:hover:not(:disabled) {
    background-color: ${getBackground};
    border-color: ${getBackground};
  }
`;
