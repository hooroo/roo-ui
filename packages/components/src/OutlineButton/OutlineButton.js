import { color, buttonStyle } from 'styled-system';
import { Button } from '..';

const getBackground = props =>
  color(props).backgroundColor || buttonStyle(props).backgroundColor;

const OutlineButton = Button.extend`
  background-color: transparent;
  border-color: ${getBackground};
  transition: none;

  &:not(:hover), &:disabled {
    color: ${getBackground};
  }

  &:hover:not(:disabled) {
    background-color: ${getBackground};
    border-color: ${getBackground};
  }
`;

OutlineButton.displayName = 'OutlineButton';

export default OutlineButton;
