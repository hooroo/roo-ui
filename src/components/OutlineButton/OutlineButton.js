import styled from '@emotion/styled';
import { color, buttonStyle, themeGet } from 'styled-system';
import get from 'lodash/get';
import { Button } from '../';

const getBackground = props =>
  get(color(props), 'backgroundColor') ||
  get(buttonStyle(props), 'backgroundColor');

const OutlineButton = styled(Button)`
  background-color: transparent;
  border-color: ${getBackground};
  transition: none;

  &:not(:hover) {
    color: ${getBackground};
  }

  &:hover:not(:disabled) {
    background-color: ${getBackground};
    border-color: ${getBackground};
  }

  &:hover:disabled {
    background-color: transparent;
  }

  &:disabled {
    color: ${themeGet('colors.greys.dusty')};
    border-color: ${themeGet('colors.greys.dusty')};
  }
`;

export default OutlineButton;
