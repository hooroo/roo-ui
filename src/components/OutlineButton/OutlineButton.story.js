import React from 'react';
import { withDocs } from 'storybook-readme';
import { boolean, select } from '@storybook/addon-knobs';

import OutlineButton from './OutlineButton';
import README from './README.md';

export default {
  title: 'Components/OutlineButton',
  decorators: [withDocs(README)],
};

export const Default = () => (
  <OutlineButton
    variant={select('Variant', ['default', 'primary'], 'default')}
    rounded={boolean('Rounded', false)}
    disabled={boolean('Disabled', false)}
  >
    Hello world
  </OutlineButton>
);

Default.story = {
  name: 'default',
};
