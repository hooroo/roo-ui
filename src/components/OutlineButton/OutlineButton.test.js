import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { mountWithTheme } from '@roo-ui/test-utils';
import { axe } from 'jest-axe';

import OutlineButton from '.';

describe('<OutlineButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(
      <OutlineButton>Click here</OutlineButton>,
      theme,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('primary', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(
        <OutlineButton variant="primary">Hello world</OutlineButton>,
        theme,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('rounded', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(
        <OutlineButton variant="rounded">Hello world</OutlineButton>,
        theme,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
