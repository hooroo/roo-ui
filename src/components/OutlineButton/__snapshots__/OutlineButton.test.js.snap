// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutlineButton /> primary renders correctly 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0.75rem 1.5rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.125em;
  -moz-letter-spacing: 0.125em;
  -ms-letter-spacing: 0.125em;
  letter-spacing: 0.125em;
  line-height: 1.35;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  border: 2px solid;
  border-color: transparent;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background-color 200ms ease-in;
  transition: background-color 200ms ease-in;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  color: white;
  background-color: #E40000;
  background-color: transparent;
  border-color: #E40000;
  -webkit-transition: none;
  transition: none;
}

.emotion-0:hover {
  background-color: #b10000;
}

.emotion-0:hover:disabled {
  background-color: #E40000;
}

.emotion-0:focus {
  box-shadow: 0 0 2px 2px #999999;
  outline: 2px solid #8DE2E0;
}

.emotion-0:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.emotion-0:not(:hover) {
  color: #E40000;
}

.emotion-0:hover:not(:disabled) {
  background-color: #E40000;
  border-color: #E40000;
}

.emotion-0:hover:disabled {
  background-color: transparent;
}

.emotion-0:disabled {
  color: #999999;
  border-color: #999999;
}

<OutlineButton
  display="inline-block"
  variant="primary"
>
  <button
    className="emotion-0 emotion-1"
    display="inline-block"
  >
    Hello world
  </button>
</OutlineButton>
`;

exports[`<OutlineButton /> renders correctly 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0.75rem 1.5rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.125em;
  -moz-letter-spacing: 0.125em;
  -ms-letter-spacing: 0.125em;
  letter-spacing: 0.125em;
  line-height: 1.35;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  border: 2px solid;
  border-color: transparent;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background-color 200ms ease-in;
  transition: background-color 200ms ease-in;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  color: white;
  background-color: #323232;
  background-color: transparent;
  border-color: #323232;
  -webkit-transition: none;
  transition: none;
}

.emotion-0:hover {
  background-color: #181818;
}

.emotion-0:hover:disabled {
  background-color: #323232;
}

.emotion-0:focus {
  box-shadow: 0 0 2px 2px #999999;
  outline: 2px solid #8DE2E0;
}

.emotion-0:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.emotion-0:not(:hover) {
  color: #323232;
}

.emotion-0:hover:not(:disabled) {
  background-color: #323232;
  border-color: #323232;
}

.emotion-0:hover:disabled {
  background-color: transparent;
}

.emotion-0:disabled {
  color: #999999;
  border-color: #999999;
}

<OutlineButton
  display="inline-block"
  variant="default"
>
  <button
    className="emotion-0 emotion-1"
    display="inline-block"
  >
    Click here
  </button>
</OutlineButton>
`;

exports[`<OutlineButton /> rounded renders correctly 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0.75rem 1.5rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.125em;
  -moz-letter-spacing: 0.125em;
  -ms-letter-spacing: 0.125em;
  letter-spacing: 0.125em;
  line-height: 1.35;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  border: 2px solid;
  border-color: transparent;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background-color 200ms ease-in;
  transition: background-color 200ms ease-in;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  background-color: transparent;
  -webkit-transition: none;
  transition: none;
}

.emotion-0:hover {
  background-color: #000;
}

.emotion-0:hover:disabled {
  background-color: #000;
}

.emotion-0:focus {
  box-shadow: 0 0 2px 2px #999999;
  outline: 2px solid #8DE2E0;
}

.emotion-0:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.emotion-0:hover:disabled {
  background-color: transparent;
}

.emotion-0:disabled {
  color: #999999;
  border-color: #999999;
}

<OutlineButton
  display="inline-block"
  variant="rounded"
>
  <button
    className="emotion-0 emotion-1"
    display="inline-block"
  >
    Hello world
  </button>
</OutlineButton>
`;
